cvat/apps/engine/cache.py

```diff
@@ -1,323 +1,569 @@
+
+            if item_data[0]:
+                item = (item_data[0], item_data[1], self._get_checksum(item_data[0].getbuffer()))
                 self._cache.set(key, item)
+            else:
+                item = (item_data[0], item_data[1], None)
 
             return item
 
-        slogger.glob.info(f'Starting to get chunk from cache: key {key}')
-        try:
-            item = self._cache.get(key)
-        except pickle.UnpicklingError:
-            slogger.glob.error(f'Unable to get item from cache: key {key}', exc_info=True)
-            item = None
-        slogger.glob.info(f'Ending to get chunk from cache: key {key}, is_cached {bool(item)}')
-
+        item = self._get(key)
         if not item:
             item = create_item()
         else:
             # compare checksum
             item_data = item[0].getbuffer() if isinstance(item[0], io.BytesIO) else item[0]
             item_checksum = item[2] if len(item) == 3 else None
-            if item_checksum != zlib.crc32(item_data):
-                slogger.glob.info(f'Recreating cache item {key} due to checksum mismatch')
+            if item_checksum != self._get_checksum(item_data):
+                slogger.glob.info(f"Recreating cache item {key} due to checksum mismatch")
                 item = create_item()
 
         return item[0], item[1]
 
-    def get_task_chunk_data_with_mime(self, chunk_number, quality, db_data):
-        item = self._get_or_set_cache_item(
-            key=f'{db_data.id}_{chunk_number}_{quality}',
-            create_function=lambda: self._prepare_task_chunk(db_data, quality, chunk_number),
-        )
+    def _get(self, key: str) -> Optional[DataWithMime]:
+        slogger.glob.info(f"Starting to get chunk from cache: key {key}")
+        try:
+            item = self._cache.get(key)
+        except pickle.UnpicklingError:
+            slogger.glob.error(f"Unable to get item from cache: key {key}", exc_info=True)
+            item = None
+        slogger.glob.info(f"Ending to get chunk from cache: key {key}, is_cached {bool(item)}")
 
         return item
 
-    def get_selective_job_chunk_data_with_mime(self, chunk_number, quality, job):
-        item = self._get_or_set_cache_item(
-            key=f'job_{job.id}_{chunk_number}_{quality}',
-            create_function=lambda: self.prepare_selective_job_chunk(job, quality, chunk_number),
-        )
+    def _has_key(self, key: str) -> bool:
+        return self._cache.has_key(key)
+
+    def _make_cache_key_prefix(
+        self, obj: Union[models.Task, models.Segment, models.Job, models.CloudStorage]
+    ) -> str:
+        if isinstance(obj, models.Task):
+            return f"task_{obj.id}"
+        elif isinstance(obj, models.Segment):
+            return f"segment_{obj.id}"
+        elif isinstance(obj, models.Job):
+            return f"job_{obj.id}"
+        elif isinstance(obj, models.CloudStorage):
+            return f"cloudstorage_{obj.id}"
```

Comment:
The fact that it can be done this way is more of a coincidence than a rule. Doing so can lead to unexpected changes in future, e.g. if a model is renamed.

Reply:
Okay, but I guess if a model is renamed then the used cache key also should be updated. Otherwise, I think it will be odd to have a renamed model `RemoteStorage` and use the `cloudstorage_<id>` key.

Reply:
Maybe, but it not follows from the model name.